packages/nuxt/src/executors/serve/serve.impl.ts

```diff
@@ -0,0 +1,90 @@
+import { ExecutorContext, joinPathFragments } from '@nx/devkit';
+import { NuxtServeExecutorOptions } from './schema';
+import { createAsyncIterable } from '@nx/devkit/src/utils/async-iterable';
+
+// Required because nuxi is ESM package.
+export function loadNuxiDynamicImport() {
+  return Function('return import("nuxi")')() as Promise<typeof import('nuxi')>;
+}
+
+export async function* nuxtServeExecutor(
+  options: NuxtServeExecutorOptions,
+  context: ExecutorContext
+) {
+  const projectRoot =
+    context.projectsConfigurations.projects[context.projectName].root;
+  yield* createAsyncIterable<{ success: boolean; baseUrl: string }>(
+    async ({ next, error }) => {
+      try {
+        const { runCommand } = await loadNuxiDynamicImport();
+        await runCommand('dev', [projectRoot], {
+          overrides: getConfigOverrides(options, context.root, projectRoot),
+        });
+        next({
+          success: true,
+          baseUrl: `${options.https ? 'https' : 'http'}://${
+            options.host ?? 'localhost'
+          }:${options.port ?? '4200'}`,
```

Comment:
I know, I know, this is not robust, but I could not find a way to get either the command output or the actual final config.

`runCommand` is `void`.

And, [`loadNuxt`](https://nuxt.com/docs/api/kit/examples#accessing-nuxt-vite-config), even though it looks like it's doing what we want it to do (get the config), it does not return the config correctly.

With a `nuxt.config.ts` like this:
```
export default defineNuxtConfig({
  srcDir: 'src',
  devtools: { enabled: true },
  devServer: {
    host: 'localhost',
    port: 4200,
  },
```

it returns:

```
nuxt: {
...
  options: {
...
    devServer: {
      https: false,
      port: 3000,
      host: undefined,
      url: 'http://localhost:3000',
      loadingTemplate: [Function: _template]
    }
  }
}
```

Reply:
1. https://github.com/unjs/citty/pull/101
2. https://github.com/nuxt/cli/pull/278
3.